shared/js/l10n.js

```diff
@@ -1374,11 +1390,14 @@
   Context.prototype.registerLocales = function(defLocale, available) {
     /* jshint boss:true */
     this.availableLocales = [this.defaultLocale = defLocale];
+    this.localeSources[defLocale] =
+      (available && (defLocale in available)) ? available[defLocale] : 'bundle';
 
     if (available) {
-      for (var i = 0, loc; loc = available[i]; i++) {
+      for (var loc in available) {
```

Comment:
Or maybe have `registerLocales` accept three arguments: `defaultLanguage`, `availableLanguages` and (optionally)`additionalLanguages`.